### PR TITLE
Add Helm Charts

### PR DESCRIPTION
**Pull request by osokin**
### Proposed changes

Add Helm Charts

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [`CONTRIBUTING`](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/blob/main/CONTRIBUTING.md) document
- [X] If applicable, I have added tests that prove my fix is effective or that my feature works
- [X] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] I have updated any relevant documentation ([`README.md`](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/blob/main/README.md) and [`CHANGELOG.md`](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/blob/main/CHANGELOG.md))

### CODE REVIEW

File: charts/nlk/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+name: nginx-loadbalancer-kubernetes
+home: https://github.com/nginxinc/nginx-loadbalancer-kubernetes
+icon: https://raw.githubusercontent.com/nginxinc/nginx-loadbalancer-kubernetes/main/nlk-logo.svg
+keywords:
+- nginx
+- loadbalancer
+- ingress
+kubeVersion: '>= 1.22.0-0'
+maintainers:
+- email: kubernetes@nginx.com
```
Comment (ciroque) [inline charts/nlk/Chart.yaml:13]: @brianehlert  is this an acceptable value for the maintainers email?

File: charts/nlk/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v2
+appVersion: 0.1.0
+description: NGINX LoadBalancer Kubernetes
```
Comment (ciroque) [inline charts/nlk/Chart.yaml:3]: Should this be "NGINX LoadBalancer for Kubernetes" which is the full name? (I am good either way...)

File: charts/nlk/templates/nlk-configmap.yaml
Comment (ciroque) [inline charts/nlk/templates/nlk-configmap.yaml:1]: We have recently added support for TLS communications which adds several fields to the ConfigMap.

I am presently of the mind to go ahead and approve / merge this PR, but we would want to follow up pretty quickly with adding the new fields.

@osokin interested in your thoughts about adding those fields now or in a follow-on PR. The [full TLS documentation](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/tree/main/docs/tls/README.md); or, for quick reference:

`tls-mode`: used to determine which of the five TLS modes will be used to communicate with the Border Servers.
`ca-certificate`: the name of the Secret that contains the Certificate Authority certificate.
`client-certificate`: the name of the Secret that contains the Client certificate.

There is another new field (`log-level`) that allows configuring the logging level.

**Review by ciroque — APPROVED** [review]
Hi @osokin , really appreciate the contribution! This is one that has been hanging out there for a while; knew it had to be done, but didn't find the motivation to actually do it.

This looks really good, I left a couple of comments about referring to support options as that has been somewhat of a sticky proposition during development. Mostly need to know if @brianehlert  is okay using that email address, or if there is another he'd prefer.

The only part is around a recent upgrade to allow TLS usage. As detailed in the file comments, I am okay with moving forward with merging this PR as long as we follow up pretty quickly with adding the new TLS fields to the ConfigMap. I'll leave the decision to you. If you choose to kick that can down the road I would ask that you create a new Issue so the work doesn't fall by the wayside.

Again, thanks so much for the contribution, this is really helpful!

File: charts/nlk/Chart.yaml (hunk 1 above)
Comment (ciroque) [reply charts/nlk/Chart.yaml:13]: Hi @osokin , we'll need to pull this email address out of here. I think for now let's use the `name` field, one each for @ciroque and @chrisakker

```
maintainers:
   name: "@ciroque"
   name: "@chrisakker"
```

Thanks!

**Comment by osokin** [comment]
Hi @ciroque, thanks for the review!  I've just updated the pull-request based on your suggestions and recommendations.

**Pull request merged**